zebra/zapi_msg.c

```diff
@@ -2084,9 +2098,11 @@ static void zread_release_label_chunk(struct zserv *client, struct stream *msg)
 static void zread_label_manager_request(ZAPI_HANDLER_ARGS)
 {
 	if (hdr->command == ZEBRA_LABEL_MANAGER_CONNECT
-	    || hdr->command == ZEBRA_LABEL_MANAGER_CONNECT_ASYNC)
+	    || hdr->command == ZEBRA_LABEL_MANAGER_CONNECT_ASYNC) {
+		if (hdr->command == ZEBRA_LABEL_MANAGER_CONNECT)
+			client->synchronous = true;
```

Comment:
Not sure why every client requesting label chunk are marked as true here.

Reply:
Actually, only synchronous clients are marked true here.

However, there has been a request to make this patch more general purpose, so that the patch can handle the PIM synchronous client too.  As part of this request, this code in zread_label_manager_request will be removed.